### DISCUSSION TITLE
Work with pack cli

### DISCUSSION THREAD

**Discussion by cxrus** [containers/podman#15576] [Q&A]
Hi team,

I followed the Building on podman section in the official document of buildpack and run buildpack with podman successfully in the local environment, which means build successfully and I can find the image via podman images command. Then I try to reproduce the same thing in the container.
I run a container, mount the docker socket, install podman, enable the podman socket via podman system service, and set DOCKER_HOST=podman.sock, I hope it should work like the local one. However, buildpack now uses podman to pull the run image and builder image but it exports the built application image to the docker daemon instead of the podman, which is really strange. Could you tell me why this case?  
I am not sure whether the podman system service command matters since the buildpack works well in the podman machine but not in the container. I cannot tell the difference between them except for the way they enable the podman socket.

**Comment by vrothberg** [comment]
Thanks for reaching out, @cxrus!

> I run a container, mount the docker socket, install podman, enable the podman socket via podman system service, and set DOCKER_HOST=podman.sock, I hope it should work like the local one. However, buildpack now uses podman to pull the run image and builder image

I assume `buildpack` is reading the `DOCKER_HOST` variable, so it makes sense that it's now using Podman rather than Docker.

>  [...] but it exports the built application image to the docker daemon instead of the podman, which is really strange. Could you tell me why this case?

Can you elaborate on that?  How are you checking whether the image is committed to Docker?

**Reply by cxrus** [reply]
Yes! To be more specific, I run a podman official container using quay.io/podman/stable in minikube, mount the docker socket in deployment.yml, enable the podman socket via podman system service, and set DOCKER_HOST=podman.sock. Then I run `pack build ...` to build a application image. In the local env, you can use `podman images` command to list all the images and you will find the application image. However, when I do it in the container, it didn't show up. But instead, I found it in the docker daemon inside the minukube which means I ssh into the minikube, run `docker images` and found that the application image is inside which is really strange.

**Comment by rhatdan** [comment]
I still don't understand fully what you are asking?  Are you using Podman as a Docker replacement?  Or are you using a Podman container to communicate with the docker.socket through a volume mount?

Could you create a simple reproducer?

**Reply by cxrus** [reply]
Yes, I should explain it in more detail. The main goal is to use podman as a docker alternative in buildpacks. The summary is that I do all the things that I have done locally in a container, and in local, it works well but not in a container. Following are things that I have done and my thoughts.

In the local, I do things following the buildpacks official documents [here](https://buildpacks.io/docs/app-developer-guide/building-on-podman/). The main part is to enable the podman socket and set the DOCKER_HOST to podman socket. Everything works fine. I can use podman images command to find the built image.

But in the container, I enabled the podman socket via podman system service command and set the DOCKER_HOST to podman socket. However, now I cannot find the built image using podman images command. And I found that the image can be found using the docker images command means it is inside the docker daemon. 

I think it is very strange since we already set DOCKER_HOST, which means all things are handled by podman now. So I should find the image inside the podman using podman images instead of inside the docker. And I verified that the buildpacks is using podman now. 

So the only difference I can tell between local setup and container setup that can cause this problem is the way of enabling the podman socket. I am not sure whether there is something wrong inside.

**Reply by rhatdan** [reply]
Podman inside of the container has its own location of image stores, which is different from that of the host.  
If you want them shared you either need to volume mount in the image store, or leak the socket from the host into the container and use podman -remote.
